### PR TITLE
Feature/create contentblock component

### LINKED ISSUE #45: Create ContentBlock component and variations
### ContentBlock

Figma: https://www.figma.com/file/Lo1ZcBOO9t4QRbX2I0PwBQ/RBB?node-id=429%3A57

Leveraging Chakra UI components, we need to build out a ContentBlock component that can accept a few parameters in order to output the different variations on the home page.

**Customization Options**

The content block should be a single Higher Order Component that allows us to add any content within it. Primarily this content will be Headings, Paragraphs, and Buttons.

As a Developer, I should be able to customize the background color that is used on an individual block and whether it is a fade, or tint.

As a Developer I should be able to customize the image that appears as the background for this block.

As a Developer I should be able to choose to have the text appear on the left, right or in the center of the block.

**Customization Examples**

__Left aligned content, black background that fades over a custom img src, Heading, Paragraph, and 2 Buttons as children of the Higher Order Component.__

_Desktop_
ContentBlock--Left--Black

__Right aligned content, white background that fades over a custom img src, Heading, Paragraph with an inline link, and 2 buttons as children of the Higher Order Component__ 

_Desktop_

ContentBlock--Right--White

__Center aligned content, color tint over a custom img src, Paragraph, and one button that spans the width of the content__

ContentBlock--Center

### Responsive

The ContentBlock should be responsive, the content will center, span the full width, and the background color will become a tint over the background image on smaller screens

_Mobile_
ContentBlock--Left--Black-mobile

### PR DESCRIPTION
**Pull request by aka-tpayne**
# Describe your PR
Create a new ContentBlock component for laying out content on various pages.

It contains several variations: 

- centered content with tinted background image
- right aligned content with tinted background image
- right aligned content with fade from side over image
- left aligned content with tinted background image
- left aligned content with fade from side over image

Related to #45 
Fixes #45 

## Pages/Interfaces that will change
Adds some examples to the index page

### Screenshots / video of changes
Desktop:
image

Mobile:
image

## Steps to test

1. Get local going and load home page

### CODE REVIEW

File: src/components/ContentBlock.js [hunk 1]
```diff
@@ -0,0 +1,254 @@
+import React from 'react';
+import { PseudoBox, Box, Image, Flex, theme } from '@chakra-ui/core';
```
Comment (ChrisBrownie55) [inline src/components/ContentBlock.js:2]: > Warning: 'Image' is defined but never used

Comment (aka-tpayne) [reply src/components/ContentBlock.js:2]: Ah, thought I got them all. I'll clean this up.

File: src/components/ContentBlock.js [hunk 2]
```diff
@@ -0,0 +1,254 @@
+import React from 'react';
+import { PseudoBox, Box, Flex, theme } from '@chakra-ui/core';
+
+const fullWidthOverlayOpacity = 0.3;
+
+const formatHexCode = hexCode => {
```
Comment (racedale) [inline src/components/ContentBlock.js:6]: Curious, what was the scenario that this utility is for?

Comment (aka-tpayne) [reply src/components/ContentBlock.js:6]: There are different ways you can pass in the color. 

Using a theme variable, it'll pull the hex from the theme config, and then convert it into the 6 digit hex if it isn't already.

Using a 3 digit hex, it'll convert it into the 6 digit hex.

This is so I can have a consistent format to apply the 2 extra digits for the alpha in the linear-gradient for the fade scenarios.

**Pull request merged**